static/app/views/automations/detail.tsx

```diff
@@ -34,13 +41,97 @@ import {useDetectorsQuery} from 'sentry/views/detectors/hooks';
 
 const AUTOMATION_DETECTORS_LIMIT = 10;
 
+const DEFAULT_CHART_PERIOD = '7d';
+
+const PAGE_QUERY_PARAMS = [
+  'pageStatsPeriod',
+  'pageStart',
+  'pageEnd',
+  'pageUtc',
+  'cursor',
+];
+
+function getDateTimeFromQuery(query: Record<string, any>): DateTimeObject {
```

Comment:
If you use `usePageFilters()` this logic will be taken care of. It's also better because it will remember the last time filter set by the user